jsk_topic_tools/src/jsk_topic_tools/transport.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from enum import Enum
+
+import rospy
+
+
+ConnectionStatus = Enum('ConnectionStatus', 'SUBSCRIBED NOT_SUBSCRIBED')
+
+
+class ConnectionBasedTransport(rospy.SubscribeListener):
+    def __init__(self, node_name):
+        super(ConnectionBasedTransport, self).__init__()
+        rospy.init_node(node_name)
```

Comment:
> I think programmer should call rospy.init_node before creating ConnectionBasedTransport.
It would be problem if user want to create several instances of ConnectionBasedTransport

I updated the commit.